.travis/before_install.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Don't source setup.sh here, because the virtualenv might not be set up yet
+
+export NUMCORES=`grep -c ^processor /proc/cpuinfo`
+if [ ! -n "$NUMCORES" ]; then
+  export NUMCORES=`sysctl -n hw.ncpu`
+fi
+echo Using $NUMCORES cores
+
+# Install dependencies
+if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+  sudo apt-get update
+  APT_INSTALL_CMD='sudo apt-get install -y --no-install-recommends'
+  $APT_INSTALL_CMD dos2unix
+
+  # Install protobuf
+  pb_version="2.6.1"
+  pb_dir="~/.cache/pb"
+  mkdir -p "$pb_dir"
+  wget -qO- "https://github.com/google/protobuf/releases/download/v$pb_version/protobuf-$pb_version.tar.gz" | tar -xz -C "$pb_dir" --strip-components 1
+  ccache -z
+  cd "$pb_dir" && ./configure && make -j${NUMCORES} && make check && sudo make install && sudo ldconfig
+  ccache -s
+
+  # Setup Python.
+  export PYTHON_DIR="/usr/bin"
+elif [ "$TRAVIS_OS_NAME" == "osx" ]; then
+  brew install ccache protobuf
+
+  # Setup Python.
+  export PYTHON_DIR="/usr/local/bin"
+  if [ "${PYTHON_VERSION}" == "python3" ]; then
+    brew install ${PYTHON_VERSION}
+  fi
+else
+  echo Unknown OS: $TRAVIS_OS_NAME
+  exit 1
+fi
+
+pip install virtualenv
+virtualenv -p "${PYTHON_DIR}/${PYTHON_VERSION}" "${HOME}/virtualenv"
```

Comment:
python3.6 recommends using `python3.6 -m venv` to create virtualenv, but I'm not sure what's the subtle differences between these couple ways. I think the current way should be fine as onnx doesn't have crazily dependencies, probably better to just add a comment here.

Reply:
This way works both for python 2 and 3. Using venv, we'd have to branch based on python version.

Reply:
++